core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java

```diff
@@ -641,6 +652,21 @@ private int getGlobalTimeout() {
       // TODO Rename setting to something like globalRpcTimeout
       return (int) gcr.getGlobalConfiguration().transport().distributedSyncTimeout();
    }
+
+   @Override
+   public void prepareForShutdown(ScopedPersistentState state) {
+      state.setProperty("uuid", localUUID.toString());
```

Comment:
Yes, that's what I have in my second set of patches, but it makes more sense to do it here.